apps/prairielearn/src/tests/instructorCourseAdminInstances.test.ts

```diff
@@ -0,0 +1,255 @@
+
+describe('Creating a course instance', () => {
+  before(async () => {
+    // Clone the course template for testing
+    await execa('git', ['-c', 'init.defaultBranch=master', 'init', '--bare', courseOriginDir], {
+      cwd: '.',
+      env: process.env,
+    });
+
+    await execa('git', ['clone', courseOriginDir, courseLiveDir], {
+      cwd: '.',
+      env: process.env,
+    });
+
+    await fs.copy(courseTemplateDir, courseLiveDir);
+
+    const execOptions = { cwd: courseLiveDir, env: process.env };
+    await execa('git', ['add', '-A'], execOptions);
+    await execa('git', ['commit', '-m', 'Initial commit'], execOptions);
+    await execa('git', ['push', 'origin', 'master'], execOptions);
+    await execa('git', ['clone', courseOriginDir, courseDevDir], { cwd: '.', env: process.env });
+
+    await helperServer.before(courseLiveDir)();
+
+    await queryAsync(sql.update_course_repo, { repo: courseOriginDir });
+  });
+
+  after(helperServer.after);
+
+  step('create a new course instance', async () => {
+    // Fetch the course instance page for the course
+    const courseInstancePageResponse = await fetchCheerio(
+      `${siteUrl}/pl/course/1/course_admin/instances`,
+    );
+
+    assert.equal(courseInstancePageResponse.status, 200);
+
+    // Create the new course instance
+    const courseInstanceCreationResponse = await fetchCheerio(
+      `${siteUrl}/pl/course/1/course_admin/instances`,
+      {
+        method: 'POST',
+        body: new URLSearchParams({
+          __action: 'add_course_instance',
+          __csrf_token: courseInstancePageResponse.$('input[name=__csrf_token]').val() as string,
+          orig_hash: courseInstancePageResponse.$('input[name=orig_hash]').val() as string,
+          short_name: 'Fa19',
+          long_name: 'Fall 2019',
+          start_access_date: '2021-01-01T00:00',
+          end_access_date: '2021-01-02T00:00',
+        }),
+      },
+    );
+
+    assert.equal(courseInstanceCreationResponse.status, 200);
+
+    // Verify that the user is redirected to the assessments page for the new course instance
+    assert.equal(
+      courseInstanceCreationResponse.url,
+      `${siteUrl}/pl/course_instance/2/instructor/instance_admin/assessments`,
+    );
+  });
+
+  step('verify course instance has the correct info', async () => {
+    const courseInstanceInfoPath = path.join(
+      courseInstancesCourseLiveDir,
+      'Fa19', // Verify that the short_name has been used as the course instance folder's name
+      'infoCourseInstance.json',
+    );
+
+    const courseInstanceInfo = JSON.parse(await fs.readFile(courseInstanceInfoPath, 'utf8'));
+
+    assert.equal(courseInstanceInfo.longName, 'Fall 2019');
+    assert.equal(courseInstanceInfo.allowAccess.length, 1);
+    assert.equal(courseInstanceInfo.allowAccess[0].startDate, '2021-01-01 00:00:00');
+    assert.equal(courseInstanceInfo.allowAccess[0].endDate, '2021-01-02 00:00:00');
```

Comment:
cc @mwest1066 the current implementation is canonicalizing to the datetime format without `T` in the middle. Do you have any preference either way? I sort of lean towards including the `T` because it's familiar from ISO strings, but maybe the average instructor would prefer not to have them.

Reply:
Great question. If we can read the date times without the T then I think this is friendlier for most people. I think most instructors have never heard of ISO 8601 😂

Reply:
SGTM! The `input_date` sproc does explicitly handle this case:

https://github.com/PrairieLearn/PrairieLearn/blob/b8cb11c28489bdad4b0e7d6b18ed6939442d4e93/apps/prairielearn/src/sprocs/input_date.sql#L16-L18